samples/snippets/src/main/java/com/example/analytics/QuickstartSample.java

```diff
@@ -16,82 +16,104 @@
-
- 2. Provide service account credentials using one of the following options:
- - set the GOOGLE_APPLICATION_CREDENTIALS environment variable, the API
- client will use the value of this variable to find the service account key
- JSON file.
-
- https://cloud.google.com/docs/authentication/production#setting_the_environment_variable
-
- OR
-
- - manually pass the path to the service account key JSON file to the API client
- by specifying the keyFilename parameter in the constructor.
- https://cloud.google.com/docs/authentication/production#passing_the_path_to_the_service_account_key_in_code
-
- To run this sample using Maven:
-   cd java-analytics-data/samples/snippets
-   mvn compile
-   mvn exec:java -Dexec.mainClass="QuickstartSample"
+To run this sample using Maven:
+  cd java-analytics-data/samples/snippets
+  mvn compile
+  mvn exec:java -Dexec.mainClass="com.example.analytics.QuickstartSample"
  */
 
-// [START analytics_data_quickstart]
+// [START google_analytics_data_quickstart]
+import com.google.analytics.data.v1beta.BetaAnalyticsDataClient;
+import com.google.analytics.data.v1beta.BetaAnalyticsDataSettings;
+import com.google.analytics.data.v1beta.DateRange;
+import com.google.analytics.data.v1beta.Dimension;
+import com.google.analytics.data.v1beta.Metric;
+import com.google.analytics.data.v1beta.Row;
 
-import com.google.analytics.data.v1alpha.AlphaAnalyticsDataClient;
-import com.google.analytics.data.v1alpha.DateRange;
-import com.google.analytics.data.v1alpha.Dimension;
-import com.google.analytics.data.v1alpha.Entity;
-import com.google.analytics.data.v1alpha.Metric;
-import com.google.analytics.data.v1alpha.Row;
-import com.google.analytics.data.v1alpha.RunReportRequest;
-import com.google.analytics.data.v1alpha.RunReportResponse;
+import com.google.analytics.data.v1beta.RunReportRequest;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import java.io.FileInputStream;
 
 public class QuickstartSample {
 
   // This is an example snippet that calls the Google Analytics Data API and runs a simple report
   // on the provided GA4 property id.
-  static void sampleRunReport(String ga4PropertyId) throws Exception {
+  static void sampleRunReport(String propertyId, String credentialsJsonPath) throws Exception {
+    /**
+     * TODO(developer): Uncomment this variable and replace with your
+     * Google Analytics 4 property ID before running the sample.
+     */
+    // propertyId = "YOUR-GA4-PROPERTY-ID";
+
+    // [START google_analytics_data_initialize]
+    /** TODO(developer): Uncomment this variable and replace with a valid path to
+     * the credentials.json file for your service account downloaded from the
+     * Cloud Console.
+     * Otherwise, default service account credentials will be derived from
+     * the GOOGLE_APPLICATION_CREDENTIALS environment variable.
+     */
+    // credentialsJsonPath = "/path/to/credentials.json";
```

Comment:
Not really best practices, most jobs running on GCP will get credentials following the ADC algorithm by fetching from the Metadata server.   (Note - I'll only ask for a change if I ask for something else)

Reply:
@lesv This is a #1 request from our users, so we decided to create a separate version of quickstart for this type of auth. I moved this to a separate quickstart sample that uses explicit JSON credentials file.